re.c

```diff
@@ -2353,8 +2360,27 @@ match_named_captures(VALUE match)
     if (NIL_P(RMATCH(match)->regexp))
         return rb_hash_new();
 
+    VALUE opt;
+    VALUE symbolize_names = 0;
+
+    rb_scan_args(argc, argv, "0:", &opt);
+
+    if (!NIL_P(opt)) {
+        static ID keyword_ids[1];
+
+        VALUE symbolize_names_val;
+
+        if (!keyword_ids[0]) {
+            keyword_ids[0] = rb_intern("symbolize_names");
```

Comment:
`rb_intern` also caches the ID, with gcc.
```suggestion
            keyword_ids[0] = rb_intern_const("symbolize_names");
```